### DISCUSSION TITLE
Two M2M fields on the same related collection: what would be the correct model?

### DISCUSSION THREAD

**Discussion by Pachat** [directus/directus#17194] [Archive]
The first collection is `Student` with the field `Name`
The second is `Ratings` with fields `Rating`, `Subject`, `Semester`.

Let's say a student has ratings every semester in the subjects history and geography.
So I configured a first M2M `Student <=> History` and a second M2M `Student <=> Geography` **both linked to `Ratings`**
That is, before saving the M2M field, uncheck the Autofill box and change the junction table name to
`Student_Eval_History` and `Student_Eval_Geography`.

Then create a student with assessment in history and geography. That works.

Still logged in with admin rights, edit a rating by clicking on it:
the following message is raised

- `[PROHIBITED] You are not authorized to access it.`

Ignore the message, edit the rating and save it. That works.

Hence my question: 

- Why this inappropriate error message?
- Is this a bug or the model's fault?

_The reason why all M2Ms are tied to the same `Ratings` collection is that since all ratings have the same structure, they fit nicely into the `Ratings` collection, and moreover, besides this MVP example , the real `Ratings` collection has many fields. Creating a different collection for each rating is extra work, and it's easier to get an average of the ratings from one collection than from several._

**Comment by Pachat** [comment]
It looks like that Directus does not accept two or more M2M fields on the same related collection.

**Reply by br41nslug** [reply]
The API allows this out of the box but in the UI you'll have to allow duplicates before it allows you to select the second.
![image](https://user-images.githubusercontent.com/9389634/212747342-3456477d-a1fa-486e-beb1-a487672ddfe2.png)

**Reply by Pachat** [reply]
Thanks for the hint. It works with 1 M2M field.
But the problem still appears when I have a second M2M_2 field on the same collection as M2M.

In the OP I have 2 M2M fields: 1 for `History` the other for `Geography` which both relate to `Eval` each via their own junction table.

And, despite allowing duplicates, it still produces the unwanted error message

      [PROHIBITED] You are not authorized to access it.

Ignoring the message, editing the rating and saving it, works.
But my users will think something is wrong and will stop on the _false_ error message.
